src/XrdHttp/XrdHttpReq.cc

```diff
@@ -2272,33 +2272,48 @@ int XrdHttpReq::PostProcessHTTPReq(bool final_) {
 
               getfhandle();
               
-              // Now parse the stat info if we still don't have it
-              if (filesize == 0) {
-                if (iovP[1].iov_len > 1) {
-                  TRACEI(REQ, "Stat for GET " << resource.c_str() 
-                           << " stat=" << (char *) iovP[1].iov_base);
-              
-                  long dummyl;
-                  sscanf((const char *) iovP[1].iov_base, "%ld %lld %ld %ld",
-                        &dummyl,
-                        &filesize,
-                        &fileflags,
-                        &filemodtime);
+              // Always try to parse response.  In the case of a caching proxy, the open
+              // will have created the file in cache
+              filectime = 0;
+              if (iovP[1].iov_len > 1) {
+                TRACEI(REQ, "Stat for GET " << resource.c_str()
+                         << " stat=" << (char *) iovP[1].iov_base);
 
-                  // As above: if the client specified a response size, we use that.
-                  // Otherwise, utilize the filesize
-                  if (!length) {
-                    length = filesize;
-                  }
+                long dummyl;
+                sscanf((const char *) iovP[1].iov_base, "%ld %lld %ld %ld %ld",
+                      &dummyl,
+                      &filesize,
+                      &fileflags,
+                      &filemodtime,
+                      &filectime);
+
+                // As above: if the client specified a response size, we use that.
+                // Otherwise, utilize the filesize
+                if (!length) {
+                  length = filesize;
                 }
-                else
-                  TRACEI(ALL, "GET returned no STAT information. Internal error?");
               }
-              
+              else {
+                TRACEI(ALL, "GET returned no STAT information. Internal error?");
+              }
+
+              std::string responseHeader;
+              if (!m_digest_header.empty()) {
+                responseHeader = m_digest_header;
+              }
+              long one;
+              if (filemodtime && XrdOucEnv::Import("XRDPFC", one)) {
```

Comment:
Are you sure you mean Import(...), not GetInt("XRDPFC")?

Reply:
Yeah, I think this is right as it matches the `Export`.  There's no obvious global to pull from.

Reply:
Yes, right ... I missed that Export puts stuff into process environment.